modules/web/src/app/shared/components/label-form/component.ts

```diff
@@ -221,16 +233,38 @@ export class LabelFormComponent implements OnChanges, OnInit, OnDestroy, Control
         this._addLabel(key, this.labels[key]);
       });
     }
-
     // Add initial label for the user.
     this._addLabel();
+    this._updateLabelsObject();
   }
 
   isRemovable(index: number): boolean {
-    return index < this.labelArray.length - 1 && !this._isInherited(Object.keys(this.labels)[index]);
+    return (
+      (index < this.labelArray.length - 1 && !this._isInherited(Object.keys(this.labels)[index])) ||
+      this.isProtectedKey(this.labelArray.controls[index].get('key').value, index)
```

Comment:
Shouldn't this be the other way around? i.e. label is removable when its _not_ protected?
```suggestion
      !this.isProtectedKey(this.labelArray.controls[index].get('key').value, index)
```